app/src/main/java/echoserver/SocketIo.java

```diff
@@ -0,0 +1,42 @@
+package echoserver;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.Socket;
+
+public class SocketIo {
+  private final Socket clientSocket;
+  public BufferedReader in;
+  public PrintWriter out;
```

Comment:
Do these need to be public? 👀

Reply:
This was another choice I made for testing purposes. I made them public so that I could test that the SocketIo constructor method created the BufferedReader and PrintWriter. Should I just rely on the tests that check that they send/receive & delete the tests that check for creation?

Reply:
Yeah, your other tests wouldn't pass without the existence of a `BufferedReader` and `PrintWriter`, so that's good enough to validate that they're getting created. FWIW, it's considered bad practice to make things public solely for the sake of testing.